packages/client/components/Dialog/Dialog.tsx

```diff
@@ -0,0 +1,29 @@
+import React from 'react'
+import * as RadixDialog from '@radix-ui/react-dialog'
+
+interface DialogProps {
+  open: boolean
+  onClose: () => void
+  children: React.ReactNode
+}
+
+const Dialog = (props: DialogProps) => {
+  const {open, onClose, children} = props
+  return (
+    <RadixDialog.Root
+      open={open}
+      onOpenChange={(newOpen) => {
+        if (!newOpen) {
+          onClose()
+        }
+      }}
```

Comment:
+1 as rule of thumb when creating a generic component, that could be reused in various places, we should allow to pass all the props, if the use case requires that. a good example of exposing prestyled, yet configurable generic components is [Dialog - shadcn/ui](https://ui.shadcn.com/docs/components/dialog) - see installation and `manual` tab, you'll see nicely typed interfaces that allow to override all sorts of things.

Reply:
@BartoszJarocki I see they store everything in one file, do you think it make sense in our case?

Reply:
no strong opinions, I think the general consensus is to have one main export from one file, let's keep them separate